### DISCUSSION TITLE
iced_lazy Responsive Widget doesn't emit Message

### DISCUSSION THREAD

**Discussion by maxslarsson** [hecrj/iced#1347] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

To render the pixels for an Image widget, I need to have a way of getting the current width and height of the Image widget. Wrapping it with the Responsive widget from iced_lazy, and using Length::Fill, fixes the issue, but for some reason this widget takes a lambda function and gives the size as an argument to the lambda. The problem is I need to save the Size of the Responsive widget since I am recalculating the pixels only when the user presses a button (and I don't want to do it every time view is called). However, since I am using pure widgets I don't have a mutable reference to self and can't save the Size. Why does the Responsive widget not just emit a Message, like Button and TextInput and most other widgets, so the update function can handle the message and rerender the pixel data? Am I misunderstanding where I should generate the pixels for an image? 

### What is the expected behavior?

The Responsive widget takes a message to produce with the size of the widget. Some ideas of what it could look like: Button or TextInput.

```rust
Responsive::new(Image::new(...)).on_resize(Message::Resize)

```

or

```rust
Responsive::new(Image::new(...), Message::Resize)
```

### Version

0.4

### Operative System

macOS

**Comment by hecrj** [comment]
`Responsive` does not emit a `Message` because, if it did, emitting a `Message` on resize would cause an `update`. This would, in turn, allow you to change the state of the `Application` and, therefore, `view` could change as well. As a result, the `Responsive` widget could have a different size and trigger another `Message`, making infinite loops in your application quite likely to happen.

If you want to cache the pixels of an `Image` after regenerating them, then I recommend you to use [interior mutability](https://doc.rust-lang.org/std/cell/index.html).
